include/roboteam_ai/stp/Tactic.h

```diff
@@ -27,23 +23,20 @@ namespace rtt::ai::stp {
          * Though this method is responsible for ensuring everything is calculated, it helps to use helpers so this
          * function doesn't become a massive hack
          */
-        TacticInfo calculateInfoForSkill();
-
-        virtual Status onInitialize() noexcept = 0;
+        virtual SkillInfo calculateInfoForSkill(TacticInfo const &info) noexcept = 0;
 
-        virtual Status onUpdate(TacticInfo const &info) noexcept = 0;
+        virtual void onInitialize() noexcept = 0;
+        virtual Status onUpdate(SkillInfo const &info) noexcept = 0;
```

Comment:
Should be Tactic Info

```suggestion
        virtual Status onUpdate(TacticInfo const &info) noexcept = 0;
```

Reply:
Also, you may want to add the status of the current skill as a param